pallets/capacity/src/lib.rs

```diff
@@ -719,11 +721,11 @@ impl<T: Config> Pallet<T> {
 			Self::get_reward_pool_for_era(era).ok_or(Error::<T>::EraOutOfRange)?;
 		reward_pool.total_staked_token = reward_pool.total_staked_token.saturating_add(*amount);
 
-		// TODO:  add boost history record for era when boost history is implemented
 		Self::set_staking_account_and_lock(staker, staking_details)?;
 		Self::set_target_details_for(staker, *target, target_details);
 		Self::set_capacity_for(*target, capacity_details);
 		Self::set_reward_pool(era, &reward_pool);
+		Self::upsert_boost_history(staker, era, *amount, true)?;
```

Comment:
I ran the weights locally on this. It bumps the boost db cost to 6 writes and 9 reads. `create_sponsored_account_with_delegation` is 7 and 10, but it is one of the highest on the chain.(`add_public_key_to_msa` is tied-ish at 7 and 8)

We just have a lot of db work that has to happen on `provider_boost`. 

I don't know of any way to reduce it, but something to be aware of.

PS, might be worth running the benchmarks on this branch so it is clear the cost of the changes.

Reply:
There is other benchmark work to do for on_initialize after adding the reward pool / era updates.  I will add this task to that issue.

Reply:
NB: This particular PR doesn't bump the provider boost branch but 1r and 1w.  It was already at 5w and 8r.